src/check/parse/Parser.zig

```diff
@@ -356,14 +365,44 @@ pub fn parseStmt(self: *Parser) ?IR.NodeStore.StatementIdx {
                 self.advance();
             }
             if (self.peek() == .UpperIdent or (qualifier != null and self.peek() == .NoSpaceDotUpperIdent)) {
+                var exposes: IR.NodeStore.TokenSpan = .{ .span = .{ .start = 0, .len = 0 } };
+                const module_name_tok = self.pos;
+                if (self.peekNext() == .KwExposing) {
+                    self.advance(); // Advance past ident
+                    self.advance(); // Advance past KwExposing
+                    self.expect(.OpenSquare) catch {
+                        return self.pushMalformed(IR.NodeStore.StatementIdx, .import_exposing_no_open);
+                    };
+                    const scratch_top = self.store.scratchTokenTop();
+                    while (self.peek() != .CloseSquare) {
+                        if (self.parseIdent()) |ident| {
+                            self.store.addScratchToken(ident);
```

Comment:
It would be nice if the `exposes` field would hold an enum along the lines of

```zig
const ExposedItem = union(enum) {
    value: Ident.Idx,
    @"type": Ident.Idx,
    custom_union: struct {
        name: Ident.Idx,
        tags: []Ident.Idx,
    },
};
```

The main benefit of `value` being distinct from `@"type"` is that we don't have to look at the first letter of the text to check if it's uppercase.

Reply:
That's what I did :-). Pretty much, and without seeing this comment